src/popup.js

```diff
@@ -121,84 +122,135 @@ async function callChatGPT(question, callback, onDone) {
-  \n\n
-  Patch of the Pull Request to review:
-  \n
-  ${patch}
-  \n\n
-  
+  Act as a code reviewer of a Pull Request, providing feedback on the code changes below. Do not introduce yourselves.
+  The change is has the following title: ${title}.
   As a code reviewer, your task is:
   - Review the code changes (diffs) in the patch and provide feedback.
   - If there are any bugs, highlight them.
   - Does the code do what it says in the commit messages?
   - Do not highlight minor issues and nitpicks.
-  - Use bullet points if you have multiple comments.`
+  - Use bullet points if you have multiple comments.
+  - Be as concise as possible
+  - Assume positive intent
+
+  You are provided with the code changes in a patch format.
+  Each patch entry has the commit message in the Subject line followed by the code changes (diffs) in a unidiff format.
+
+  \n\n
+  Patch of the code change to review:
+  \n
+  ${patch}
+  \n\n
+
+  Additionally, a description was given to help you assist in understand why these changes were made.
+  \n
+  \n
+  Description of the code change to review in a markdown format:
+  \n
+  ${context}
+  \n
+  \n`
 
   callChatGPT(
     prompt,
     (answer) => {
       document.getElementById('result').innerHTML = converter.makeHtml(answer)
     },
     () => {
-      chrome.storage.session.set({ [org + repo + pr]: document.getElementById('result').innerHTML })
+      chrome.storage.session.set({ [diffPath]: document.getElementById('result').innerHTML })
       inProgress(false)
     }
   )
 }
 
+
 async function run() {
 
   let tab = (await chrome.tabs.query({ active: true, currentWindow: true }))[0];
   let prUrl = document.getElementById('pr-url')
   prUrl.textContent = tab.url
 
+  let diffPath
+  let provider = ''
+  let error = null
   let tokens = tab.url.split('/')
-  if (tokens[2] !== 'github.com' || tokens[5] !== 'pull') {
-    document.getElementById('result').innerHTML = 'Please open a specific PR on github.com'
-    inProgress(false, true, false)
-    await new Promise(r => setTimeout(r, 4000));
+  let context = ''
+  let title = tab.title
+
+  if (tokens[2] === 'github.com') {
+    provider = 'GitHub'
+  }
+  else if (tokens[2] === 'gitlab.com') {
+    provider = 'GitLab'
+  }
+  else {
+    error = 'Only github.com or gitlab.com are supported.'
+  }
+
+  let contextRaw = await fetch (tab.url).then((r) => r.text())
+  const contextDom = parse(contextRaw);
+
+  if (provider === 'GitHub' && tokens[5] === 'pull') {
+    // The path towards the patch file of this change
+    diffPath = `https://patch-diff.githubusercontent.com/raw/${tokens[3]}/${tokens[4]}/pull/${tokens[6]}.patch`;
+    // The description of the author of the change
+    context = contextDom.querySelector('.markdown-body').textContent;
+  }
+  else if (provider === 'GitLab' && tab.url.includes('/-/merge_requests/')) {
+    // The path towards the patch file of this change
+    diffPath = tab.url + '.patch';
```

Comment:
nice! I double checked that this works even if you're viewing a different 'tab' of the MR, appending `.patch` still loads the patch